File: Daily Exercises/Mar 2021/2021-03-01 : 90. Subsets II/90. Subsets II.java
class Solution {
    List<List<Integer>> ans=new ArrayList<List<Integer>>();
    List<Integer> path=new ArrayList<Integer>();
    
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        dfs(nums,0);
        return ans;
    }
    public void dfs(int[] nums,int index){
        if (index > nums.length) return;
        ans.add(new ArrayList<>(path));
        for (int i = index; i < nums.length; ++i) {
            if (i != index && nums[i] == nums[i - 1]) continue;
            path.add(nums[i]);
            dfs(nums,i+1);
            path.remove(path.size()-1);
        }
    }
}

File: Daily Exercises/Mar 2021/2021-03-01 : 90. Subsets II/90. Subsets II.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        
        def dfs(index):
            if path not in ans:
                ans.append(deepcopy(path))
            for i in range(index, len(nums)):
                if i>index and nums[i]==nums[i-1]:
                    continue
                path.append(nums[i])
                dfs(i + 1)
                path.pop()

        ans = []
        path=[]
        nums.sort()
        dfs(0)
        return ans
